llvm/lib/Target/DirectX/DXILOpLowering.cpp

```diff
@@ -739,6 +739,50 @@ class OpLowerer {
     });
   }
 
+  [[nodiscard]] bool lowerIsFPClass(Function &F) {
+    IRBuilder<> &IRB = OpBuilder.getIRB();
+    Type *RetTy = IRB.getInt1Ty();
+
+    return replaceFunction(F, [&](CallInst *CI) -> Error {
+      IRB.SetInsertPoint(CI);
+      SmallVector<Value *> Args;
+      Value *Fl = CI->getArgOperand(0);
+      Args.push_back(Fl);
+
+      dxil::OpCode OpCode;
+      Value *T = CI->getArgOperand(1);
+      auto *TCI = dyn_cast<ConstantInt>(T);
+      switch (TCI->getZExtValue()) {
+      case FPClassTest::fcInf:
+        OpCode = dxil::OpCode::IsInf;
```

Comment:
I'm assuming because we can use `IsInf` here that means its fine that we have `let intrinsics = [IntrinSelect<int_dx_isinf>];` defined and for the other `dxil::OpCodes` we should be able to add their intrinsic mapping when we implement them?

Reply:
Yes, both the dx_isinf intrinsic and the llvm.is.fpclass intrinsic can lower to the IsInf DXIL op. Their lowerings are independent of each other. 

Once the other dx_ intrinsics are defined, they can be mapped to their respective DXIL ops using `intrinsics = [IntrinSelect<int_dx_*>]`.